Sources/SwiftKueryMySQL/MySQLConnection.swift

```diff
@@ -217,9 +238,22 @@ public class MySQLConnection: Connection {
     /// - Parameter query: The query to execute.
     /// - Parameter onCompletion: The function to be called when the execution of the query has completed.
     public func execute(query: Query, onCompletion: @escaping ((QueryResult) -> ())) {
-        if let preparedStatement = prepareStatement(query, onCompletion: onCompletion) {
-            preparedStatement.execute(onCompletion: onCompletion)
-            preparedStatement.release()
+        prepareStatement(query) { result in
+            guard let statement = result.asPreparedStatement else {
+                if let error = result.asError {
+                    return self.runCompletionHandler(.error(QueryError.databaseError(error.localizedDescription)), onCompletion: onCompletion)
+                }
+                return self.runCompletionHandler(.error(QueryError.databaseError("Unable to prepare statement")), onCompletion: onCompletion)
+            }
+            self.execute(preparedStatement: statement) { result in
+                if result.asResultSet == nil {
+                    self.release(preparedStatement: statement) { _ in
+                        return self.runCompletionHandler(result, onCompletion: onCompletion)
+                    }
+                    return
+                }
+                return self.runCompletionHandler(result, onCompletion: onCompletion)
```

Comment:
Who is responsible for releasing `statement` in this case?  Is it the `MySQLResultFetcher` (during deinit or when `done()` called)?  It might be worth adding a comment above this line to explain how we ensure that `PreparedStatement` artefacts are properly cleaned up.

Reply:
Comment added